### PR TITLE
test environment: update aruba/cucumber deps to latest stable

### PR DESCRIPTION
**Pull request by mroth**
while Aruba is still not shipping 1.0 (see #15), making me a bit nervous about the state of the project, it's been long enough that I want to try to get the tests running on latest stable anyhow, so we're in a cleaner state to be prepared, and to work towards starting to get CI working in Windows environments as well.

### CODE REVIEW

File: features/step_definitions/scmpuff_steps.rb [hunk 1]
```diff
@@ -73,13 +73,13 @@
       system("git commit -m.")
     end
   end
-  FileUtils.cp_r MOCK, current_dir
+  FileUtils.cp_r MOCK, expand_path(".")
```
Comment (mroth) [inline features/step_definitions/scmpuff_steps.rb]: This is a silly and outdated rule, Hound, but I don't care enough to change your settings right now.

**Comment by mroth** [comment]
Closer, down to only one failing test...

**Comment by mroth** [comment]
I've narrowed this final test down and it appears to be caused by a regression in Aruba. I'm going to update the step with some inline documentation that shows what is happening (and possible hacky workarounds), then file an issue with Aruba and see if anyone has thoughts.

**Comment by mroth** [comment]
Issue filed upstream as cucumber/aruba#432.  Fingers crossed for a better solution.

**Comment by jdelStrother** [comment]
Urgh, wish I'd spotted this PR 2 hours ago. I also started work on upgrading cucumber & aruba & fixing tests.

WDYT to this approach to working around https://github.com/cucumber/aruba/issues/432 ?

https://github.com/jdelStrother/scmpuff/commit/e1691b682b1d5cf6426ed6a0773707295893d2fd

**Comment by mroth** [comment]
I'm not certain... I believe shells often have subtle differences in how they handle certain things when they detect they are being run in "interactive mode."

As a sidetone, a few years ago I also got about 60-70% progress into writing an entire working replacement for Aruba for the Go-native "godog" cucumber framework.  It had a lot of other niceties such as being able to run all the tests in isolated environments that can be run entirely concurrently, making the tests go very fast. One day I'd still like to pick that up again, but haven't had time and it's not relevant to any of the other OSS projects I've been working on nor my day job... (the repo is currently private but let me know if you're interested in it.)

**Comment by jdelStrother** [comment]
> I'm not certain... I believe shells often have subtle differences in how they handle certain things when they detect they are being run in "interactive mode."

It is still running the shell interactively (https://github.com/jdelStrother/scmpuff/commit/e1691b682b1d5cf6426ed6a0773707295893d2fd#diff-98ae2f02b44c253b557b5dd380b1f41f84c353c580b15443c0f25c3e3d1b4c58R109) - the main change is automatically adding an "exit" and the end of the list of commands, followed by waiting for the shell to actually exit.

> As a sidenote, a few years ago I also got about 60-70% progress into writing an entire working replacement for Aruba for the Go-native "godog" cucumber framework. It had a lot of other niceties such as being able to run all the tests in isolated environments that can be run entirely concurrently, making the tests go very fast. One day I'd still like to pick that up again, but haven't had time and it's not relevant to any of the other OSS projects I've been working on nor my day job... (the repo is currently private but let me know if you're interested in it.)

Sounds interesting, but like you think I might be too time constrained to actually do much with it 😕

**Comment by jdelStrother** [comment]
FWIW I rebased this branch & added a `command.wait` to fix those few failing tests: https://github.com/mroth/scmpuff/compare/master...jdelStrother:aruba-update

I tried to update Aruba to the latest version, but versions >= 1 no longer allow you to set the root directory, which is frustrating. Not sure what the best way around that is.

**Pull request closed without merging**